src/main/java/bio/terra/janitor/service/pubsub/TrackedResourceSubscriber.java

```diff
@@ -0,0 +1,65 @@
+package bio.terra.janitor.service.pubsub;
+
+import bio.terra.generated.model.CreateResourceRequestBody;
+import bio.terra.janitor.app.configuration.TrackResourcePubsubConfiguration;
+import bio.terra.janitor.common.exception.InvalidMessageException;
+import bio.terra.janitor.service.janitor.JanitorService;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.cloud.pubsub.v1.AckReplyConsumer;
+import com.google.cloud.pubsub.v1.MessageReceiver;
+import com.google.cloud.pubsub.v1.Subscriber;
+import com.google.pubsub.v1.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/** Subscribes track resource message and update Janitor record. */
+@Component
+public class TrackedResourceSubscriber {
+  private final TrackResourcePubsubConfiguration trackResourcePubsubConfiguration;
+  private final JanitorService janitorService;
+
+  @Autowired
+  TrackedResourceSubscriber(
+      TrackResourcePubsubConfiguration trackResourcePubsubConfiguration,
+      JanitorService janitorService) {
+    this.trackResourcePubsubConfiguration = trackResourcePubsubConfiguration;
+    this.janitorService = janitorService;
+  }
+
+  public void initialize() {
+    if (!trackResourcePubsubConfiguration.isPubsubEnabled()) {
+      return;
+    }
+
+    createTrackedResourceSubscriber(
+        trackResourcePubsubConfiguration.getProjectId(),
+        trackResourcePubsubConfiguration.getSubscription());
+  }
+
+  private void createTrackedResourceSubscriber(String projectId, String subscriptionId) {
```

Comment:
seems a little funny to pull this into a separate method when it's the end of the function and there's nothing else to the initialize function.